compiler/rustc_const_eval/src/interpret/projection.rs

```diff
@@ -202,11 +202,6 @@ where
         // see https://github.com/rust-lang/rust/issues/93688#issuecomment-1032929496.)
         // So we just "offset" by 0.
         let layout = base.layout().for_variant(self, variant);
-        if layout.abi.is_uninhabited() {
-            // `read_discriminant` should have excluded uninhabited variants... but ConstProp calls
-            // us on dead code.
-            throw_inval!(ConstPropNonsense)
-        }
```

Comment:
Please replace this by a comment explaining why we do allow uninhabited variants, and pointing to https://github.com/rust-lang/rust/issues/115145.